lib/handlers/subscribe.js

```diff
@@ -130,13 +124,25 @@ function completeSubscribe (err) {
     return done(err)
   }
 
-  broker.emit('subscribe', packet.subscriptions, client)
+  if (!packet.restore) {
+    broker.emit('subscribe', packet.subscriptions, client)
+  }
 
   if (packet.messageId) {
     write(client, new SubAck(packet, granted), done)
   } else {
     done()
   }
+
+  var persistence = broker.persistence
+  var topics = packet.subscriptions.map(function (s) { return s.topic })
```

Comment:
Done @GavinDmello 
Can you also check https://github.com/mcollina/aedes-persistence-redis/pull/31 ?